store/store.go

```diff
@@ -1326,7 +1345,66 @@ type PaymentInformation struct {
 	Methods                []*PaymentMethod `json:"methods"`
 }
 
+type storeCustomer struct {
+	LatestTosDate     string `json:"latest_tos_date"`
+	AcceptedTosDate   string `json:"accepted_tos_date"`
+	LatestTosAccepted bool   `json:"latest_tos_accepted"`
+	HasPaymentMethod  bool   `json:"has_payment_method"`
+}
+
+// ReadyToBuy returns nil if the user's account has accepted T&Cs and has a payment method registered, and an error otherwise
+func (s *Store) ReadyToBuy(user *auth.UserState) error {
+	if user == nil {
+		return ErrInvalidCredentials
+	}
+
+	reqOptions := &requestOptions{
+		Method: "GET",
+		URL:    s.customersMeURI,
+		Accept: jsonContentType,
+	}
+	resp, err := s.doRequest(s.client, reqOptions, user)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		var customer storeCustomer
+		dec := json.NewDecoder(resp.Body)
+		if err := dec.Decode(&customer); err != nil {
+			return err
+		}
+		if !customer.LatestTosAccepted {
+			return ErrTosNotAccepted
+		}
+		if !customer.HasPaymentMethod {
+			return ErrNoValidPaymentMethod
+		}
+		// TODO Check if the user has a payment method registered (once the store API provides this info)
+		return nil
+	case http.StatusNotFound:
+		// Likely because user has no account registered on the pay server
+		return fmt.Errorf("cannot get customer details: server says no account exists")
+	case http.StatusUnauthorized:
+		return ErrInvalidCredentials
+	default:
+		var errors storeErrors
+		dec := json.NewDecoder(resp.Body)
+		if err := dec.Decode(&errors); err != nil {
+			return err
+		}
+		details := ""
+		if len(errors.Errors) > 0 && errors.Errors[0].Message != "" {
+			details = ": " + errors.Errors[0].Message
+		}
+		return fmt.Errorf("cannot get customer details: unexpected HTTP code %d%s", resp.StatusCode, details)
```

Comment:
As Michael pointed out, let's please just do `return &errors` here if len(errors) > 0, and have an Error method that creates a nice message. Also, in those cases let's please not mention the HTTP code. We have a high-level error message which is more descriptive than the HTTP code.